### ISSUE TITLE
Sonarr unable to connect to indexer after Mono upgrade to 5.20.1.34 

### ISSUE THREAD

**Issue by dietevil** [SynoCommunity/spksrc#4206]
_For new Package Requests, see the guidelines_  

Issue appears to be related to `mono`

### Setup
_Package Name:_ mono
_Package Version:_ 5.20.1.34

_NAS Model:_ DS418
_NAS Architecture:_ Realtek RTD1296 SoC
_DSM version:_ 6.2.3-25426 Update 2

### Expected behavior
Sonarr should be able to connect to usenet indexer.

### Actual behavior
Sonarr fails to connect to usenet indexer.

### Steps to reproduce
_1._ Update mono
_2._ Restart NAS
_3._ Test indexer in Sonarr

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
N/A
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
20-10-5 21:17:33.1|Info|RssSyncService|Starting RSS Sync
20-10-5 21:17:33.1|Warn|FetchAndParseRssService|No available indexers. check your configuration.
20-10-5 21:17:33.1|Info|DownloadDecisionMaker|No results found
20-10-5 21:17:33.1|Info|RssSyncService|RSS Sync Completed. Reports found: 0, Reports grabbed: 0
20-10-5 21:17:58.6|Warn|Newznab|Unable to connect to indexer

[v2.0.0.5344] System.Net.WebException: Error: SecureChannelFailure (Authentication failed, see inner exception.): 'https://nzbs.in/api?t=caps&apikey=(removed) ---> System.Net.WebException: Error: SecureChannelFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:0f000078:elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE
  at /spksrc/spk/mono/work-aarch64-6.1/mono-5.20.1.34/external/boringssl/crypto/ec/ec.c:835
  at Mono.Btls.MonoBtlsContext.ProcessHandshake () [0x00038] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x000a1] in <c5f4e317158343a5916a953327f92540>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x000ff] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.AsyncProtocolRequest.StartOperation (System.Threading.CancellationToken cancellationToken) [0x0008b] in <c5f4e317158343a5916a953327f92540>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) [0x00252] in <c5f4e317158343a5916a953327f92540>:0 
  at Mono.Net.Security.MonoTlsStream.CreateStream (System.Net.WebConnectionTunnel tunnel, System.Threading.CancellationToken cancellationToken) [0x00126] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x001ba] in <c5f4e317158343a5916a953327f92540>:0 
   --- End of inner exception stack trace ---
  at System.Net.WebConnection.CreateStream (System.Net.WebOperation operation, System.Boolean reused, System.Threading.CancellationToken cancellationToken) [0x0021a] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebConnection.InitConnection (System.Net.WebOperation operation, System.Threading.CancellationToken cancellationToken) [0x00141] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebOperation.Run () [0x0009a] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.WebCompletionSource`1[T].WaitForCompletion () [0x00094] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <c5f4e317158343a5916a953327f92540>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x00016] in <c5f4e317158343a5916a953327f92540>:0 
  at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x0011b] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
   --- End of inner exception stack trace ---
  at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x001f6] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Common.Http.Dispatchers.FallbackHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x0003c] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Common.Http.HttpClient.ExecuteRequest (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookieContainer) [0x0007e] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Common.Http.HttpClient.Execute (NzbDrone.Common.Http.HttpRequest request) [0x000b2] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Common.Http.HttpClient.Get (NzbDrone.Common.Http.HttpRequest request) [0x00007] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider.FetchCapabilities (NzbDrone.Core.Indexers.Newznab.NewznabSettings indexerSettings) [0x0009a] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider+<>c__DisplayClass4_0.<GetCapabilities>b__0 () [0x00000] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 
  at NzbDrone.Common.Cache.Cached`1[T].Get (System.String key, System.Func`1[TResult] function, System.Nullable`1[T] lifeTime) [0x0006a] in <faeb8209b6ff4c15aabf93dc8df43c9f>:0 
  at NzbDrone.Core.Indexers.Newznab.NewznabCapabilitiesProvider.GetCapabilities (NzbDrone.Core.Indexers.Newznab.NewznabSettings indexerSettings) [0x00046] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 
  at NzbDrone.Core.Indexers.Newznab.Newznab.get_PageSize () [0x0000c] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 
  at NzbDrone.Core.Indexers.Newznab.Newznab.GetRequestGenerator () [0x0000b] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 
  at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].TestConnection () [0x00007] in <4d3cc4e5f7ed4ed2814d83bdaf94809e>:0 

20-10-5 21:17:58.7|Warn|NzbDroneErrorPipeline|Invalid request Validation failed: 
 -- Unable to connect to indexer, check the log for more details
```

**Comment by elmerenges** [comment]
Same here.

Package Name: mono
Package Version: 5.20.1.34-15

NAS Model: DS218
NAS Architecture: Realtek RTD1296 SoC
DSM version: 6.2.3-25426 Update 2

**Comment by silv3r23** [comment]
same:

Package Name: mono
Package Version: 5.20.1.34-15

NAS Model: DS418j
NAS Architecture: Realtek RTD1293 SoC
DSM version: DSM 6.2.3-25426 Update 2

**Comment by omearamd** [comment]
Same issue Synology DS418 Realtek RTD1296
Package Name: mono
Package Version: 5.20.1.34-15

**Comment by limdul** [comment]
Mono SSL issue

"Mono.Btls.MonoBtlsException: Ssl error:0f000078:elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE"

this may help https://github.com/SynoCommunity/spksrc/issues/3666

**Comment by omearamd** [comment]
> Mono SSL issue
> 
> "Mono.Btls.MonoBtlsException: Ssl error:0f000078:elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE"
> 
> this may help #3666

Thanks, similar symptom but my system was working with Mono 5.18, the issue started for me upgrading to Mono 5.20.  As posters in that thread mention, you can't downgrade Mono without downgrading Radarr and Sonarr, and even then Radarr and Sonarr force update to newer Mono version.  I don't want to start tampering with version numbers and mess up the repo dependencies...  Hoping this is fixed, as it seems to be new with 5.20.1.34-15

**Comment by st1650** [comment]
Same issue with Jackett
Tried both with the SynoPKG and just directly running the Jackett mono release via mono --debug JackettConsole.exe

Exception (ettv): Ssl error:0f000078:elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE at /spksrc/spk/mono/work-aarch64-6.1/mono-5.20.1.34/external/boringssl/crypto/ec/ec.c:835: An error occurred while sending the request

Mono 5.20.1.34

What about trying to make a dotnetcore package (and maybe installing via entware) and running those .net apps in "native" mode rather than mono ? or do we have to wait for DSM 7.x ?

**Comment by st1650** [comment]
I fixed it by rolling back to mono 5.8. Newer versions of Jackett and Sonarr still work with Mono 5.8

**Comment by ymartin59** [comment]
OK but 5.8 is really old.
With "some hope", I have just published another Mono update with recent OpenSSL and fix for certificates as 5.20.1.34-16

**Comment by omearamd** [comment]
Thanks for update, I installed but still get the POINT_IS_NOT_ON_CURVE error after update and reboot.

**Comment by st1650** [comment]
Well here is how I do it and believe it or not, I'm able to run the latest version of Sonarr 3 and Jackett.
1) Delete the mono package from the DSM package manager.
2) Manually delete the actual mono folder (ie /volume1/@appstore/mono)
3) Manually install the older mono 5.8 package from the syno community
4) Download the latest releases from Sonarr ( Sonarr.phantom-develop.3.0.4.986.linux.tar.gz ) & Jackett (Jackett.Binaries.Mono.tar.gz) via github
5) run those exec ideally via tmux - 
/volume1/@appstore/mono/bin/mono --debug Jackett/JackettConsole.exe
/volume1/@appstore/mono/bin/mono --debug Sonarr/Sonarr.exe
6) Make some kind of script to launch them at boot.
It's not elegant but at least it'll work until someone fix the newer mono packages.

**Comment by ppouliot** [comment]
Fwiw, tried with the lastest mono package update 5.20.1.34-16 and this still spits the same errors about being unable to connect to the indexers.

**Comment by Stanzilla** [comment]
Yup, same here

**Comment by kirilan** [comment]
same error here

**Comment by sqwal** [comment]
Same same ;-)

**Comment by Dirkle** [comment]
Reporting this is broken for DS1515+ and the [avoton-x64](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model#avoton-x64) architecture. Downgrading to [mono v5.8.0.108-11](https://synocommunity.com/package/mono) and [sonarr v20200409-16](https://synocommunity.com/package/nzbdrone) as mentioned in #3666 has worked for me. Didn't need to go as extensive a downgrade as st1650 or others here mentioned. Just:
1. backup sonarr (through the sonarr UI),
1. uninstall sonarr
1. uninstall mono
1. download  [mono v5.8.0.108-11](https://synocommunity.com/package/mono) and [sonarr v20200409-16](https://synocommunity.com/package/nzbdrone)
1. Manually install mono
1. Manually install sonarr
1. Follow nzbdrone [Backup and Restore](https://nzbdrone.readthedocs.io/Backup-and-Restore/) steps to restore sonarr

**Comment by tabacco** [comment]
This seems to have struck me DS418J this week - Guessing maybe a held-back release was re-tagged for that architecture at some point, since this seems to have been an issue for quite a while?

**Comment by plyngby** [comment]
Also happened to my DS918+

**Comment by Taloth** [comment]
@ymartin59 Sonarr Dev here.

> OK but 5.8 is really old.

Yeah, and Sonarr v3 has mono 5.18 as supported minimum so downgrade to 5.8 isn't really a workaround.

> With "some hope", I have just published another Mono update with recent OpenSSL and fix for certificates as 5.20.1.34-16

Mono doesn't use openssl. It uses boringssl and it's compiled part of mono itself. (`/spksrc/spk/mono/work-aarch64-6.1/mono-5.18.0.240/external/boringssl/crypto/ec/ec.c:835`)
There have been compiler issues with arm64 in the past, so I suspect the same is happening here with syno's aarch64 crosscompiler. I'm not entirely sure if mono compiles boringssl as a separate .so file, if so then comparing it to an aarch64 mono package compiled without the syno compiler stack.

PS: Small piece of test code https://gist.github.com/Taloth/1e7702be5e1d38675aac099bc40e957d which can be compiled with `mcs` and run with `mono` on the affected devices. So it can be tested without running Sonarr/Radarr.

**Comment by CerveloFellow** [comment]
Getting the same error on my DS418.  I compiled and ran the test code above from Taloth, and attached the output if it helps.

[error.log](https://github.com/SynoCommunity/spksrc/files/6062675/error.log)

**Comment by mreid-tt** [comment]
Think we've tracked down the problem with Mono and made a new PR #5604 for it. Thus far testing seems to be working well. Until it is published, those willing can download from the [Artifacts](https://github.com/SynoCommunity/spksrc/actions/runs/4158324234) section of the build for testing.

**Comment by mreid-tt** [comment]
Hey guys, I don't know if this is the same situation but I've been tracking an issue with mono and certificate validation errors after updating to Mono 5.20.1.34-18. There may be a few options to try if you are willing to help test:

1. Install a new version of mono from the [PR](https://github.com/SynoCommunity/spksrc/pull/5604) I've been testing which addresses the certificate verification errors. Download your applicable arch version from the [Artifacts](https://github.com/SynoCommunity/spksrc/actions/runs/4158324234) section of the build
2. Install a dotnet based version of Sonarr v4 from the [PR](https://github.com/SynoCommunity/spksrc/pull/5524) I've created. Download your applicable arch version from the [Artifacts](https://github.com/SynoCommunity/spksrc/actions/runs/4157812718) section of the build

If you are unsure of which architecture you have please refer to this tutorial page -- [What kind of CPU does my Synology NAS have?](https://kb.synology.com/en-me/DSM/tutorial/What_kind_of_CPU_does_my_NAS_have) Let me know if either of these helps to resolve this issue.

**Comment by mreid-tt** [comment]
@dietevil, on looking into this issue further, for `armv8` archs on DSM 6.x this is a known issue with Mono versions greater than 5.8.0.  This was noted in https://github.com/SynoCommunity/spksrc/issues/3946#issuecomment-612356178. If you are still running on DSM 6, you can downgrade your Mono version by uninstalling the current version and re-installing v5.8.0.108-11 from the repository. Alternately you can manually download that version at [6.1 rtd1296](https://packages.synocommunity.com/mono/11/mono.v11.f15047%5Brtd1296-armada37xx-aarch64%5D.spk).

If you are running on DSM 7 you should be able to install Mono version for [aarch64-7.1](https://github.com/SynoCommunity/spksrc/suites/11076561292/artifacts/563553567) which will be published to the repository shortly.

I'll be closing this issue in a few days if there are no objections.

**Comment by mreid-tt** [comment]
Closing this issue as a known bug which is unable to be fixed. Workarounds have been presented above.

**Comment by mreid-tt** [comment]
Hi @dietevil, I have some great news to share with you! We've recently published version 6.12.0.182 (#5677) which addresses the SSL store problems on aarch64 platforms (#3666) that you reported. This update should be available for download shortly. We would be very interested in hearing your feedback on this release, as we are confident it should resolve the issue you've been experiencing. Thank you for your patience and for bringing this to our attention!

**Comment by bootsieke** [comment]
Hi @mreid-tt , would this update also work on a DS411 with 88f628x architecture? 

When I tried updating the working 5.8 version about 1.5 year ago I got similar issues:

System.Security.Cryptography.CryptographicException: Unable to decode certificate. ---> System.DllNotFoundException: /volume2/@appstore/mono/lib/../lib/libmono-btls-shared.so

**Comment by mreid-tt** [comment]
@bootsieke, Mono is not my area of expertise but there is a version for your platform for version 6.12 ([6.1 88f628x](https://packages.synocommunity.com/mono/20/mono.v20.f15047%5B88f628x%5D.spk)). You can try it to see if by upgrading it resolves the issue. If not you can always roll back to version 5.8 with a manual rollback to the older package ([6.1 88f628x](https://packages.synocommunity.com/mono/11/mono.v11.f15047%5B88f628x%5D.spk)). There is a method in the Wiki to [downgrade a package without uninstalling (Advanced)](https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#downgrade-a-package-without-uninstalling-advanced) if you're into that.

**Comment by hyundaimumpsimus** [comment]
@mreid-tt, the version 6.12 for the 628x is no longer available. Do you know were I could get it?

**Comment by mreid-tt** [comment]
> @mreid-tt, the version 6.12 for the 628x is no longer available. Do you know were I could get it?

My guess is that it was removed. Based on the release notes for https://github.com/SynoCommunity/spksrc/pull/5677, the ARMv5 builds were disabled because they don't contain a TLS library. As such, the link I referred to was probably an unintended upload. The version of Mono supporting this architecture (v5.8) is still present however.

@hgy59 is my interpretation of the above correct?
